crates/sui-framework/sources/object.move

```diff
@@ -114,6 +128,13 @@ module sui::object {
         delete_impl(bytes)
     }
 
+    /// Delete the object and it's `UID` and return the proof of deletion - `DeletedUID`.
+    public fun delete_with_proof(id: UID): DeletedUID {
```

Comment:
Could you add a test of this being used? I'm wondering whether the ID Leak verifier is currently smart enough to recognise that a UID passed to `delete_with_proof` will definitely be `delete`d.

Reply:
Would a simple "new" + "delete_with_proof" work?

Reply:
Added a full test with a separate module and `next_tx` calls.